tock/employees/models.py

```diff
@@ -38,6 +40,23 @@ class EmployeeGrade(models.Model):
     def __str__(self):
         return '{0} - {1} (Starting: {2})'.format(self.employee, self.grade, self.g_start_date)
 
+    def get_grade(end_date, user):
+        """Gets grade information based on a date and user. Original use of
+        function is to append correct grade information to each
+        TimecardObject."""
+        emp_grd_objects = EmployeeGrade.objects.filter(
+                Q(employee = user)
+                & Q(g_start_date__lte = end_date)
+                ).all().aggregate(Max('g_start_date'))['g_start_date__max']
+
+        if emp_grd_objects:
```

Comment:
Yep, good catch on the duplicate queries. Simplified it (and learned about all of the methods available for QuerySet objects!) via https://github.com/18F/tock/pull/461/commits/88892abbb8d89ffff56172691e75a31243abf187. So you don't have to go digging, here it is:

```
    def get_grade(end_date, user):
        """Gets grade information based on a date and user. Original use of
        function is to append correct grade information to each
        TimecardObject."""
        queryset = EmployeeGrade.objects.filter(
                Q(employee = user)
                & Q(g_start_date__lte = end_date)
                ).all()
        if queryset:
            return queryset.latest('g_start_date')
        else:
            return None
```